cinema/views.py

```diff
@@ -112,27 +136,52 @@
 
         return MovieSessionSerializer
 
+    def perform_create(self, serializer):
+        if not self.request.user.is_staff:
+            raise PermissionDenied(
+                "You do not have permission to perform this action."
+            )
+        serializer.save()
+
+    def perform_update(self, serializer):
+        if not self.request.user.is_staff:
+            raise PermissionDenied(
+                "You do not have permission to perform this action."
+            )
+        serializer.save()
+
+    def perform_destroy(self, instance):
+        if not self.request.user.is_staff:
+            raise PermissionDenied(
+                "You do not have permission to perform this action."
+            )
+        instance.delete()
+
 
 class OrderPagination(PageNumberPagination):
     page_size = 10
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    GenericViewSet,
+):
     queryset = Order.objects.prefetch_related(
-        "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
+        "tickets__movie_session__movie",
+        "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
 
-    def get_queryset(self):
-        return Order.objects.filter(user=self.request.user)
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
 
-    def get_serializer_class(self):
-        if self.action == "list":
-            return OrderListSerializer
+    def get_permissions(self):
+        if self.action == "create":
+            return [IsAuthenticated()]
```

Comment:
The `get_permissions` method in `OrderViewSet` should ensure that the `IsAuthenticated` permission is correctly applied when creating an order without overriding other permissions. This could lead to unauthorized access.